src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesProducer.java

```diff
@@ -75,6 +75,9 @@ public KNN80DocValuesProducer(DocValuesProducer delegate, SegmentReadState state
                 continue;
             }
             List<String> engineFiles = KNNCodecUtil.getEngineFiles(knnEngine.getExtension(), field.name, state.segmentInfo);
+            if (engineFiles.isEmpty()) {
+                continue;
+            }
```

Comment:
why we need this?

Reply:
Line 81 assumes that engine files will always be available, that is not the case anymore.

Reply:
Created new PR #2147 to address this change